File: JavaScript/8js 1/arrayMadness.js
/*
|--------------------------------------------------------------------------
| SpeedCode #2 - Array Madness
|--------------------------------------------------------------------------
|
*/

function arrayMadness(a, b) {
   return (
      a.map((i) => i ** 2).reduce((x, y) => x + y) > b.map((i) => i ** 3).reduce((x, y) => x + y)
   );
}


function arrayMadness(a, b) {
   return a.reduce((pre, val) => pre + val ** 2, 0) > b.reduce((pre, val) => pre + val ** 3, 0);
}


function arrayMadness(a, b) {
   const x = (arr, n) => arr.map((v) => v ** n).reduce((a, b) => a + b);
   return x(a, 2) > x(b, 3);
}


console.log(arrayMadness([4, 5, 6], [1, 2, 3]), true);
console.log(arrayMadness([5, 6, 7], [4, 5, 6]), false);
console.log(arrayMadness([4, 5, 6], [3, 4, 5]), false);
console.log(arrayMadness([3, 4, 5], [2, 3, 4]), false);
console.log(arrayMadness([1, 2, 3, 4, 5], [2, 2, 2, 2, 2, 2, 1, 1, 1]), true);

File: Python/8py 1/array_madness.py
"""
|--------------------------------------------------------------------------
| SpeedCode #2 - Array Madness
|--------------------------------------------------------------------------
|
"""

def array_madness(a, b):
   return sum([i**2 for i in a]) > sum([i**3 for i in b])


array_madness = lambda a,b:sum(_*_ for _ in a)>sum(__**3 for __ in b)


def array_madness(a,b):
   return sum(map(lambda a: a ** 2, a)) > sum(map(lambda b: b ** 3, b))


print(array_madness([4, 5, 6], [1, 2, 3]),True)
print(array_madness([1, 2, 3], [4, 5, 6]),False)